src/Google/GoogleHelper.php

```diff
@@ -0,0 +1,1340 @@
+		return [
+			'US' => 'USD', // United States
+		];
+	}
+
+	/**
+	 * Get an array of Google Merchant Center supported countries for promotions.
+	 *
+	 * @return string[]
+	 */
+	public function get_mc_promotion_supported_countries(): array {
+		return array_keys( $this->get_mc_promotion_supported_countries_currencies() );
+	}
+
+	/**
+	 * Get an array of Google Merchant Center supported languages (ISO 639-1).
+	 *
+	 * WooCommerce Languages -> https://translate.wordpress.org/projects/wp-plugins/woocommerce/
+	 * Google Supported Languages -> https://support.google.com/merchants/answer/160637?hl=en
+	 * ISO 639-1 -> https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes
+	 *
+	 * @return array
+	 */
+	public function get_mc_supported_languages(): array {
+		// Repeated values removed:
+		// 'pt', // Brazilian Portuguese
+		// 'zh', // Simplified Chinese*
+
+		return [
+			'ar' => 'ar', // Arabic
+			'cs' => 'cs', // Czech
+			'da' => 'da', // Danish
+			'nl' => 'nl', // Dutch
+			'en' => 'en', // English
+			'fi' => 'fi', // Finnish
+			'fr' => 'fr', // French
+			'de' => 'de', // German
+			'he' => 'he', // Hebrew
+			'hu' => 'hu', // Hungarian
+			'id' => 'id', // Indonesian
+			'it' => 'it', // Italian
+			'ja' => 'ja', // Japanese
+			'ko' => 'ko', // Korean
+			'el' => 'el', // Modern Greek
+			'nb' => 'nb', // Norwegian (Norsk Bokmål)
+			'nn' => 'nn', // Norwegian (Norsk Nynorsk)
+			'no' => 'no', // Norwegian
+			'pl' => 'pl', // Polish
+			'pt' => 'pt', // Portuguese
+			'ro' => 'ro', // Romanian
+			'ru' => 'ru', // Russian
+			'sk' => 'sk', // Slovak
+			'es' => 'es', // Spanish
+			'sv' => 'sv', // Swedish
+			'th' => 'th', // Thai
+			'zh' => 'zh', // Traditional Chinese
+			'tr' => 'tr', // Turkish
+			'uk' => 'uk', // Ukrainian
+			'vi' => 'vi', // Vietnamese
+		];
+	}
+
+	/**
+	 * Get whether the country is supported by the Merchant Center.
+	 *
+	 * @param string $country Country code.
+	 *
+	 * @return bool True if the country is in the list of MC-supported countries.
+	 */
+	public function is_country_supported( string $country ): bool {
+		return array_key_exists(
+			strtoupper( $country ),
+			$this->get_mc_supported_countries_data()
+		);
+	}
+
+	/**
+	 * Maps the ID of the Merchant Center supported locations to their ISO 3166-1 codes.
+	 */
+	protected function map_location_id_to_code(): void {
+		$this->location_id_code_map = [];
+
+		$countries = $this->get_mc_supported_countries_data();
+		foreach ( $countries as $country ) {
+			$this->location_id_code_map[ $country['id'] ] = $country['code'];
+			if ( isset( $country['subdivisions'] ) ) {
+				foreach ( $country['subdivisions'] as $subdivision ) {
+					$this->location_id_code_map[ $subdivision['id'] ] = $subdivision['code'];
+				}
+			}
+		}
+	}
+
+	/**
+	 * @param int $id
+	 *
+	 * @return string|null
+	 */
+	public function find_location_code_by_id( int $id ): ?string {
+		if ( ! isset( $this->location_id_code_map ) ) {
+			$this->map_location_id_to_code();
+		}
```

Comment:
I had another thought about this function `find_location_code_by_id`. From what I understand it's main purpose is to be used by the multi country targeting for campaigns. Would it make more sense though to make this function specifically work for countries (and maybe a second one for regions)?

Otherwise if I call it with `21141` the code would be `DE`. I can see by the name that's it's Delaware and not Germany. But I don't really have a way to tell if that code is a country or a region.

Reply:
> What do you think if we move if ( ! isset( $this->location_id_code_map ) ) into map_location_id_to_code function.

I thought about that but then that would be changing the functionality of the `map` method because when I call it I expect it to "map" the latest updates and not return a cached version. This isn't of concern in this class though since we only call it one place and the data is hardcoded but I think it's good practice to leave it as is. Otherwise, I can change the method to `get_mapped_location_id_code` and have it return a cached version.

Reply:
Okay, but to be clear I need to change the method's name and return value as well. Updated in https://github.com/woocommerce/google-listings-and-ads/pull/1282/commits/9111f9414eec2b137f68772fd5f0f6e482d9317b